src/terms/xc.jl

```diff
@@ -25,32 +26,44 @@ function Base.show(io::IO, xc::Xc)
     print(io, "Xc($fun$fac)")
 end
 
-function (xc::Xc)(::PlaneWaveBasis{T}) where {T}
+function (xc::Xc)(basis::PlaneWaveBasis{T}) where {T}
     isempty(xc.functionals) && return TermNoop()
+    # Charge density for non-linear core correction
+    if any(use_nlcc, basis.model.atoms)
+        ρcore = core_density_superposition(basis)
+    else
+        ρcore = nothing
+    end
     functionals = map(xc.functionals) do fun
         # Strip duals from functional parameters if needed
         newparams = convert_dual.(T, parameters(fun))
         change_parameters(fun, newparams; keep_identifier=true)
     end
     TermXc(convert(Vector{Functional}, functionals),
            convert_dual(T, xc.scaling_factor),
-           T(xc.potential_threshold))
+           T(xc.potential_threshold),
+        ρcore   )
 end
 
 struct TermXc{T} <: TermNonlinear where {T}
     functionals::Vector{Functional}
     scaling_factor::T
     potential_threshold::T
+    ρcore::Union{Nothing, Array{T,4}}
 end
 
-@views @timing "ene_ops: xc" function ene_ops(term::TermXc, basis::PlaneWaveBasis{T},
-                                              ψ, occupation; ρ, τ=nothing, kwargs...) where {T}
+function xc_potential_real(term::TermXc, basis::PlaneWaveBasis{T}, ψ, occupation; ρ, τ=nothing) where {T}
```

Comment:
not completely sure this refactoring is useful?

Reply:
Ah, I see below why you did this. Just use total_local_potential?

Reply:
It's a bit difficult to see how best to pull the xc potential out from that, what would you do?

Reply:
You have to iterate through the ops and get those corresponding to the kpoints with the appropriate spin, see how `total_local_potential` does it

Reply:
I misspoke earlier: it doesn't seem like you can call total_local_potential directly, you have to reproduce the logic I think

Reply:
> you have to reproduce the logic I think

Sounds like the best solution for now, I agree. Let's see what comes out of #784. Could be useful to solve this one as well.

Reply:
For this, I'd need access to the `Hamiltonian` struct, but this isn't passed to `compute_forces`. Without re-evaluating the Hamiltonian, I'd need to change the interface for all `compute_forces` methods. Is that reasonable, or am I missing something?